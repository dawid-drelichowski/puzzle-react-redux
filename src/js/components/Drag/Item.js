import React, {PropTypes} from 'react';

export default class extends React.Component {
  static propTypes = {
    field: PropTypes.number.isRequired
  };
  render() {
    const props = this.props;
    return <li draggable="true" data-field={props.field}/>;
  }
}
