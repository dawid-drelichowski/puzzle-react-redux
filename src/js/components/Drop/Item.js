import React, {PropTypes} from 'react';

export default class extends React.Component {
  static propTypes = {
    field: PropTypes.number.isRequired,
    currentlyDragged: PropTypes.number,
    onDrop: PropTypes.func.isRequired
  };
  static defaultProps = {
    currentlyDragged: 0
  };
  state = {dropped: false};
  onDragOver(event) {
    if (this.isValidDragElement()) {
      event.preventDefault();
    }
  }
  onDrop(event) {
    if (this.isValidDragElement()) {
      event.preventDefault();
      this.setState({dropped: true});
      this.props.onDrop(this.props.field);
    }
  }
  render() {
    let props = {};

    switch (this.state.dropped) {
      case true:
        props = {
          'data-field': this.props.field
        };
        break;
      default:
        props = {
          'data-valid': this.props.field,
          onDragOver: event => this.onDragOver(event),
          onDrop: event => this.onDrop(event)
        };
        break;
    }
    return <li {...props}></li>;
  }
  isValidDragElement() {
    return this.props.field === this.props.currentlyDragged;
  }
}
