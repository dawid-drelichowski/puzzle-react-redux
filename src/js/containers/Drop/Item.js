import {connect} from 'react-redux';
import Creators from 'actions/index';
import Item from 'components/Drop/Item';

export default connect(
  state => {
    return {currentlyDragged: state.currentlyDragged};
  },
  dispatch => {
    return {
      onDrop: index => {
        dispatch(Creators.dropped(index));
      }
    };
  }
)(Item);
