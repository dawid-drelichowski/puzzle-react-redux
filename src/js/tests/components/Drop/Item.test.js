import React from 'react';
import Item from 'components/Drop/Item';
import renderer from 'react-test-renderer';

describe('Drop item component', () => {
  function getComponent(field, dropped = false) {
    return renderer.create(
      <Item field={field} dropped={dropped}/>
    );
  }

  it('should be 2nd item not dropped', () => {
    expect(getComponent(2).toJSON()).toMatchSnapshot();
  });

  it('should be 4th item not dropped', () => {
    expect(getComponent(4).toJSON()).toMatchSnapshot();
  });

  it('should be 2nd item dropped', () => {
    expect(getComponent(2, true).toJSON()).toMatchSnapshot();
  });

  it('should be 4th item dropped', () => {
    expect(getComponent(4, true).toJSON()).toMatchSnapshot();
  });
});
