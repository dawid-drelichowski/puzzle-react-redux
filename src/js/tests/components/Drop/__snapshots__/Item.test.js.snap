// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drop item component should be 2nd item dropped 1`] = `
<li
  data-field={2}
/>
`;

exports[`Drop item component should be 2nd item not dropped 1`] = `
<li
  data-valid={2}
/>
`;

exports[`Drop item component should be 4th item dropped 1`] = `
<li
  data-field={4}
/>
`;

exports[`Drop item component should be 4th item not dropped 1`] = `
<li
  data-valid={4}
/>
`;
